src/main/java/com/minecolonies/colony/Colony.java

```diff
@@ -160,6 +163,19 @@ private void readFromNBT(NBTTagCompound compound)
             }
         }
 
+        //todo update fields from tileEntity.
+        // Fields
+        NBTTagList fieldTagList = compound.getTagList(TAG_FIELDS, NBT.TAG_COMPOUND);
+        for (int i = 0; i < fieldTagList.tagCount(); ++i)
+        {
+            NBTTagCompound      fieldCompound = fieldTagList.getCompoundTagAt(i);
+            Field               f             = Field.createFromNBT(this, fieldCompound);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'f' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)